hibernate-core/src/main/java/org/hibernate/query/sqm/tree/expression/SqmLiteralEmbeddableType.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.query.sqm.tree.expression;
+
+import org.hibernate.metamodel.model.domain.EmbeddableDomainType;
+import org.hibernate.query.hql.HqlInterpretationException;
+import org.hibernate.query.hql.spi.SemanticPathPart;
+import org.hibernate.query.hql.spi.SqmCreationState;
+import org.hibernate.query.sqm.NodeBuilder;
+import org.hibernate.query.sqm.SemanticQueryWalker;
+import org.hibernate.query.sqm.SqmExpressible;
+import org.hibernate.query.sqm.tree.SqmCopyContext;
+import org.hibernate.query.sqm.tree.domain.SqmPath;
+import org.hibernate.query.sqm.tree.select.SqmSelectableNode;
+import org.hibernate.type.BasicType;
+import org.hibernate.type.StandardBasicTypes;
+
+/**
+ * Represents a reference to an embeddable type as a literal.
+ *
+ * @author Marco Belladelli
+ */
+public class SqmLiteralEmbeddableType<T>
+		extends AbstractSqmExpression<T>
+		implements SqmSelectableNode<T>, SemanticPathPart {
+	final EmbeddableDomainType<T> embeddableDomainType;
+
+	public SqmLiteralEmbeddableType(
+			EmbeddableDomainType<T> embeddableDomainType,
+			NodeBuilder nodeBuilder) {
+		//noinspection unchecked
+		super(
+				(SqmExpressible<? super T>) nodeBuilder.getTypeConfiguration()
```

Comment:
> this depends on whether we want to allow literals for embeddables that are not polymorphic

Yes, exactly. We now could have access to the discriminator type through `MetadataImplementor`, but if we wish to keep supporting "simple" embeddables we would still need to resolve to the basic `Class` type (not that using `type(simple_embeddable)` worked before as it just rendered a class-typed `SqmLiteral`, so unsupporting this would be a "breaking" change).